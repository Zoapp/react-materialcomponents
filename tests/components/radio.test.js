import React from "react";
import renderer from "react-test-renderer";
import Radio from "@libs/components/radio";

describe("components/Radio", () => {
  it("can have an id", () => {
    const component = renderer.create(
      <Radio
        id="1"
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can have a label", () => {
    const component = renderer.create(
      <Radio
        label="foo"
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can have be disabled", () => {
    const component = renderer.create(
      <Radio
        disabled
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can be checked", () => {
    const component = renderer.create(
      <Radio
        checked
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can have a name", () => {
    const component = renderer.create(
      <Radio
        name="bar"
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
