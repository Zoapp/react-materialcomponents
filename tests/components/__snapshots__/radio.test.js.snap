// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Radio can be checked 1`] = `
<div
  className="mdc-radio"
>
  <input
    className="mdc-radio__native-control"
    defaultChecked="checked"
    id="0.bg7tlkk6zla"
    name={null}
    type="radio"
  />
  <div
    className="mdc-radio__background"
  >
    <div
      className="mdc-radio__outer-circle"
    />
    <div
      className="mdc-radio__inner-circle"
    />
  </div>
</div>
`;

exports[`components/Radio can have a label 1`] = `
<div
  className="mdc-form-field"
>
  <div
    className="mdc-radio"
  >
    <input
      className="mdc-radio__native-control"
      id="0.6mbaxsc48ru"
      name={null}
      type="radio"
    />
    <div
      className="mdc-radio__background"
    >
      <div
        className="mdc-radio__outer-circle"
      />
      <div
        className="mdc-radio__inner-circle"
      />
    </div>
  </div>
  <label
    htmlFor="0.6mbaxsc48ru"
  >
    foo
  </label>
</div>
`;

exports[`components/Radio can have a name 1`] = `
<div
  className="mdc-radio"
>
  <input
    className="mdc-radio__native-control"
    id="0.k9842a5l1es"
    name="bar"
    type="radio"
  />
  <div
    className="mdc-radio__background"
  >
    <div
      className="mdc-radio__outer-circle"
    />
    <div
      className="mdc-radio__inner-circle"
    />
  </div>
</div>
`;

exports[`components/Radio can have an id 1`] = `
<div
  className="mdc-radio"
>
  <input
    className="mdc-radio__native-control"
    id="1"
    name={null}
    type="radio"
  />
  <div
    className="mdc-radio__background"
  >
    <div
      className="mdc-radio__outer-circle"
    />
    <div
      className="mdc-radio__inner-circle"
    />
  </div>
</div>
`;

exports[`components/Radio can have be disabled 1`] = `
<div
  className="mdc-radio"
>
  <input
    className="mdc-radio__native-control"
    disabled="disabled"
    id="0.89mvvclrkgt"
    name={null}
    type="radio"
  />
  <div
    className="mdc-radio__background"
  >
    <div
      className="mdc-radio__outer-circle"
    />
    <div
      className="mdc-radio__inner-circle"
    />
  </div>
</div>
`;
